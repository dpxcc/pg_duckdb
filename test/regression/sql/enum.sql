CREATE TYPE mood AS ENUM ('happy', 'sad', 'neutral');
create table tbl (a mood);

insert into tbl select 'happy';
insert into tbl select 'neutral';
insert into tbl select 'sad';

select * from tbl;

drop table tbl;
drop type mood;
