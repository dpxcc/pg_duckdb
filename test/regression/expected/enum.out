CREATE TYPE mood AS ENUM (
	'sad',
	'neutral',
	'happy'
);
create table tbl (a mood);
insert into tbl select 'happy';
insert into tbl select 'neutral';
insert into tbl select 'sad';
select * from tbl;
    a    
---------
 happy
 neutral
 sad
(3 rows)

--select * from tbl where a = 'sad'; -- returns `sad`
--select * from tbl where a > 'neutral'; --- returns `happy`
ALTER TYPE mood ADD VALUE 'a-bit-happy' BEFORE 'happy';
ALTER TYPE mood ADD VALUE 'very-happy' AFTER 'happy';
ALTER TYPE mood ADD VALUE 'very-sad' BEFORE 'sad';
insert into tbl values ('very-sad'), ('very-happy'), ('a-bit-happy');
--select * from tbl where a > 'neutral'; --- returns `a-bit-happy`, `happy` and `very-happy`;
select * from tbl;
      a      
-------------
 happy
 neutral
 sad
 very-sad
 very-happy
 a-bit-happy
(6 rows)

drop table tbl;
drop type mood;
